docs/pages/world/modules/keysintable.mdx

```diff
@@ -0,0 +1,239 @@
+import { Callout } from "nextra/components";
+
+import { CollapseCode } from "../../../components/CollapseCode";
+
+# Keys in Table Module
+
+[The `KeysInTable` module](https://github.com/latticexyz/mud/tree/main/packages/world-modules/src/modules/keysintable) tracks which keys are used with a specific table.
+This is useful if you need to be able to iterate through all the records of a table onchain.
+
+<Callout type="warning" emoji="💰">
+It costs additional gas for every write to the table affected.
+If you just need this functionality offchain it is best to look at the data offchain instead.
+
+Also, this is a _generic_ solution.
+You can usually save gas by implementing a solution that is specific to your use case, using [hooks](/store/store-hooks).
+
+</Callout>
+
+## Deployment
+
+It can be deployed multiple times, each time for a different table.
+For example, here is a `mud.config.ts` that deploys it for the `Tasks` table in [the React template](https://github.com/latticexyz/mud/tree/main/templates/react).
+
+<CollapseCode>
+
+```typescript filename="mud.config.ts" copy showLineNumbers {2,17-32}
+import { defineWorld } from "@latticexyz/world";
+import { resourceToHex } from "@latticexyz/common";
+
+export default defineWorld({
+  namespace: "app",
+  tables: {
+    Tasks: {
+      schema: {
+        id: "bytes32",
+        createdAt: "uint256",
+        completedAt: "uint256",
+        description: "string",
+      },
+      key: ["id"],
+    },
+  },
+  modules: [
+    {
+      root: true,
+      artifactPath: "@latticexyz/world-modules/out/KeysInTableModule.sol/KeysInTableModule.json",
+      args: [
+        {
+          type: "bytes32",
+          value: resourceToHex({
+            type: "table",
+            namespace: "app",
+            name: "Tasks",
+          }), // end of value
+        }, // end of arg
+      ], // end of args
+    }, // end of module
+  ], // end of modules
+});
+```
+
+</CollapseCode>
+
+<details>
+
+<summary>Explanation</summary>
+
+```typescript
+import { resourceToHex } from "@latticexyz/common";
+```
+
+We need to provide `KeysInTable` with the resource ID for the table.
+
+```typescript
+  modules: [
+    {
+```
+
+To install modules using `mud.config.ts`, you specify `modules:` which contains a list of module defintions.
+
+```typescript
+      root: true,
+      artifactPath: "@latticexyz/world-modules/out/KeysInTableModule.sol/KeysInTableModule.json",
+      args: [
+```
+
+- `artifactPath`, a link to the compiled JSON file for the module.
+- `root`, whether to install the module with [root namespace permissions](/world/systems#root-systems) or not.
+- `args` the module arguments.
+
+```typescript
+        {
+          type: "bytes32",
+          value:
+```
+
+The `args` list contains structures.
+Each structure has two fields:
+
+- `type`, the [Solidity data type](https://docs.soliditylang.org/en/latest/types.html)
+- `value`, the value
+
+```typescript
+resourceToHex({
```

Comment:
why do we maintain indentation in code blocks above but not here? I think we should pick one or the other

Reply:
Oops, I meant to keep it.